### ISSUE TITLE
Terraform destroy does nothing

### ISSUE THREAD

**Issue by pdomineaux** [mehdilaruelle/vault-aws-gitlab-secret#1]
Hello,
Thank you for that useful workflow.

I have an issue.
The terraform plan/apply is applied and the resources are created as expected.
But if I continue the pipeline and press the manual button for the terraform destroy the resources still exist:

```bash
$ terraform destroy -auto-approve
Changes to Outputs:
  - db_engine            = "mysql" -> null
  - db_name              = "web" -> null
  - db_user              = "admin" -> null
  - vault_path_db_rotate = "web-db/rotate-root/mysql" -> null
You can apply this plan to save these new output values to the Terraform
state, without changing any real infrastructure.
Destroy complete! Resources: 0 destroyed.
Cleaning up project directory and file based variables
Job succeeded
```

**Comment by pdomineaux** [comment]
Issue fixed on my side.

It was due to the fact that I was trying to use this project mixed with a [gitlab managed terraform state](https://docs.gitlab.com/ee/user/infrastructure/iac/terraform_state.html) that was conflicting with the initial state.

I've fixed it by using this command in my pipeline :

`terraform init -reconfigure -backend-config="address= ....
`

Everything is now working as expected, and the tfstate file is available to my pipeline.
So vault/aws/gitlab-secret/gitlab-managed-terraform-state in one project.

Thanks for the good work and for sharing the knowledge.
